vlasovsolver/cpu_trans_pencils.hpp

```diff
@@ -97,6 +106,69 @@ struct setOfPencils {
       path.push_back(pathIn);
    }
 
+   void binPencils(const std::unordered_set<CellID>& targetCells) {
+      bins.resize(N);
+
+      // Loop over pencils
+      for (uint i = 0; i < N; ++i) {
+         bins[i] = i;
+         binsCells[i] = {};
+
+         for (auto id = ids.begin() + idsStart[i]; id < ids.begin() + idsStart[i] + lengthOfPencils[i]; ++id) {
+            if (*id && targetCells.contains(*id)) {
+               binsCells[i].insert(*id);
+            }
+         }
+      }
+
+      // Super ugly!
+      // TODO If anyone knows how to do this more efficiently feel free to fix it
+      std::set<uint> binsToDelete;
+      do {
+         binsToDelete.clear();
+         for (auto bin1 = binsCells.begin(); bin1 != binsCells.end(); ++bin1) {
+            if (binsToDelete.contains(bin1->first)) {
+               continue;
+            }
+
+            auto& cells1 = bin1->second;
```

Comment:
this is now target cells of bin1

Reply:
or rather, this is any source cells in bin1 which are also a target cell for any pencil. (Let's add a comment!)